tsconfig.json

```diff
@@ -1,9 +1,19 @@
 {
   "compilerOptions": {
-    "target": "es2017",
+    "allowSyntheticDefaultImports": true,
+    "allowUnreachableCode": false,
+    "allowUnusedLabels": false,
+    "alwaysStrict": true,
+    "esModuleInterop": true,
     "module": "commonjs",
     "moduleResolution": "node",
-    "esModuleInterop": true,
-    "strict": true
+    "noImplicitAny": true,
+    "noImplicitReturns": true,
+    "noImplicitThis": true,
+    "noUnusedLocals": true,
+    "noUnusedParameters": true,
+    "pretty": true,
+    "strict": true,
+    "target": "es2017",
```

Comment:
it would be nice (but a bit of effort) to go all in:

```json
    "strictNullChecks": true,
    "strictFunctionTypes": true,
    "strictPropertyInitialization": true,
```

Reply:
ah my mistake. when browsing through, i was wondering where the return types were and forgot its a lint rule rather than a compilation option. didn't realise the rest were turned on.